lib/src/signed_video_h26x_sign.c

```diff
@@ -590,10 +590,30 @@ signed_video_set_recurrence_interval(signed_video_t *self, int recurrence)
 
   svi_rc status;
   SVI_TRY()
+    SVI_THROW_IF_WITH_MSG(recurrence < RECURRENCE_ALWAYS, SVI_NOT_SUPPORTED,
+        "Recurrence interval < RECURRENCE_ALWAYS");
     SVI_THROW_IF_WITH_MSG(
-        recurrence < RECURRENCE_MIN, SVI_NOT_SUPPORTED, "Invalid recurrence interval");
+        recurrence > MAX_GOP_LENGTH, SVI_NOT_SUPPORTED, "Recurrence interval > MAX_GOP_LENGTH");
     self->recurrence = recurrence;
   SVI_CATCH()
   SVI_DONE(status)
+
+  return svi_rc_to_signed_video_rc(status);
+}
+
+#ifdef SV_UNIT_TEST
+SignedVideoReturnCode
+signed_video_set_recurrence_offset(signed_video_t *self, int offset)
+{
+  if (!self) return SV_INVALID_PARAMETER;
+
+  svi_rc status;
+  SVI_TRY()
+    SVI_THROW_IF_WITH_MSG(offset >= self->recurrence, SVI_NOT_SUPPORTED, "Invalid offset");
+    self->recurrence_offset = offset;
+  SVI_CATCH()
+  SVI_DONE(status)
```

Comment:
Maybe no need for TRY/CATCH here since it is only one try. The check is a sanity check.